src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -338,7 +338,7 @@ public void getTopEntities(ADTask adTask, ActionListener<String> internalHCListe
                     dataEndTime,
                     detectorInterval
                 );
-            if (adTask.getDetector().isMultiCategoryDetector()) {
+            if (adTask.getDetector().hasMultipleCategories()) {
```

Comment:
Question : what happens if one wishes to register many categories (say for post processing) and want to use only one -- so wouldn't this create an issue?

Reply:
The code is to issue different queries for top entities depending on there are one categorical field or multiple ones. For postprocessing there should be a different code path.

Reply:
Ok.